source/common/config/tls_context_json.cc

```diff
@@ -40,8 +40,10 @@ void TlsContextJson::translateCommonTlsContext(
   translateTlsCertificate(json_tls_context, *common_tls_context.mutable_tls_certificates()->Add());
 
   auto* validation_context = common_tls_context.mutable_validation_context();
-  validation_context->mutable_trusted_ca()->set_filename(
-      json_tls_context.getString("ca_cert_file", ""));
+  if (validation_context->has_trusted_ca()) {
```

Comment:
Shouldn't we be guarding this on the JSON object containing "ca_cert_file"? (Like "verify_certificate_hash")

Reply:
We should indeed.  That's what I get for uploading on my way out the door :-P

plus side, no cavities?